crates/proto/src/tcp/tcp_client_stream.rs

```diff
@@ -105,19 +106,29 @@ impl<S: AsyncRead + AsyncWrite + Send> Stream for TcpClientStream<S> {
 
 // TODO: create unboxed future for the TCP Stream
 /// A future that resolves to an TcpClientStream
-pub struct TcpClientConnect(
-    Box<Future<Item = TcpClientStream<TokioTcpStream>, Error = ProtoError> + Send>,
-);
+pub struct TcpClientConnect<S>(Box<Future<Item = TcpClientStream<S>, Error = ProtoError> + Send>);
 
-impl Future for TcpClientConnect {
-    type Item = TcpClientStream<TokioTcpStream>;
+impl<S> Future for TcpClientConnect<S> {
+    type Item = TcpClientStream<S>;
     type Error = ProtoError;
 
     fn poll(&mut self) -> Poll<Self::Item, Self::Error> {
         self.0.poll()
     }
 }
 
+#[cfg(any(feature = "tokio-compat", test))]
```

Comment:
Won't this annotation prevent tests from compiling on platforms that don't support tokio?

Reply:
This is probably correct. It also point later out that we don’t have any test coverage for an alternative executor to tokio. I’ll file an issue with that.

@chunyingw since this is to support your use case, I’m happy to merge as is, but you may want to validate that it compiles where you need it to.

Reply:
I do have local codes  which is running the test cases in tcp_stream.rs and tcp_client_stream.rs with fuchsia's executor. However I cannot used the codes as is. Ex. I need to create a timer instance manually in order to make tokio_timer::timeout work.
I am actually thinking we can make the test cases as procedure macro or other format in order to reuse it for different executors.

Reply:
Would you like me to merge as is or wait?

Reply:
It could be merged.
I will have test cases on fuchsia side at this moment. Once I find a way to share the test codes for tokio and fuchsia executors, I will upstream.